edn_format/edn_parse.py

```diff
@@ -78,23 +79,25 @@ def p_map(p):
     if len(terms) % 2 != 0:
         raise EDNDecodeError('Even number of terms required for map')
     # partition terms in pairs
+    terms = list(terms)
```

Comment:
We can avoid this line by changing the one below not to use ranges:
```python
p[0] = ImmutableDict(dict([[terms[i], terms[i+1]] for i in range(0, len(terms), 2)]))
```

Reply:
How would you feel about changing `ImmutableDict`'s `__init__` function to take a sequence of tuples? That way we can avoid copying the dict twice, but to do so we'll need to detect the type in `__init__`.

Reply:
Nevermind, because `dict` can handle them, we don't need to change `ImmutableDict` at all. I'll update the PR in a few minutes.